assets/samplesheet.csv

```diff
@@ -1,3 +1,14 @@
-sample,fastq_1,fastq_2
-SAMPLE_PAIRED_END,/path/to/fastq/files/AEG588A1_S1_L002_R1_001.fastq.gz,/path/to/fastq/files/AEG588A1_S1_L002_R2_001.fastq.gz
-SAMPLE_SINGLE_END,/path/to/fastq/files/AEG588A4_S4_L003_R1_001.fastq.gz,
+species,file,taxonomy,shallow,broad,mode,uniprot,mcl_test
+Diacronema_lutheri,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Diacronema_lutheri-test-proteome.fasta,Haptophyta,eukaryota_odb10,eukaryota_odb10,proteins,true,false
+Entamoeba_histolytica,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Entamoeba_histolytica-test-proteome.fasta,Amoebozoa,eukaryota_odb10,eukaryota_odb10,proteins,true,false
+Giardia_intestinalis,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Giardia_intestinalis-test-proteome.fasta,Fornicata,eukaryota_odb10,eukaryota_odb10,proteins,true,false
+Loa_loa,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Loa_loa-test-proteome.fasta,Opisthokonta,nematoda_odb10,eukaryota_odb10,proteins,true,false
+Lokiarchaeota_archaeon,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Lokiarchaeota_archaeon-test-proteome.fasta,Archaea,archaea_odb10,eukaryota_odb10,proteins,true,false
+Porphyridium_purpureum,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Porphyridium_purpureum-test-proteome.fasta,Rhodophyta,eukaryota_odb10,eukaryota_odb10,proteins,true,false
+Saccharomyces_cerevisiae,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Saccharomyces_cerevisiae-test-proteome.fasta,Opisthokonta,ascomycota_odb10,eukaryota_odb10,proteins,true,false
+Thecamonas_trahens,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Thecamonas_trahens-test-proteome.fasta,Apusomonadida,eukaryota_odb10,eukaryota_odb10,proteins,true,false
+Toxoplasma_gondii,/home/ubuntu/environment/github/phylorthology/assets/nf-test/full-data/Toxoplasma_gondii-test-proteome.fasta,Alveolata,coccidia_odb10,eukaryota_odb10,proteins,true,false
+Entamoeba_histolytica,/home/ubuntu/environment/github/phylorthology/assets/nf-test/test-data/Entamoeba_histolytica-test-proteome.fasta,Amoebozoa,eukaryota_odb10,eukaryota_odb10,proteins,true,true
+Loa_loa,/home/ubuntu/environment/github/phylorthology/assets/nf-test/test-data/Loa_loa-test-proteome.fasta,Opisthokonta,nematoda_odb10,eukaryota_odb10,proteins,true,true
+Toxoplasma_gondii,/home/ubuntu/environment/github/phylorthology/assets/nf-test/test-data/Toxoplasma_gondii-test-proteome.fasta,Alveolata,eukaryota_odb10,eukaryota_odb10,proteins,true,true
+Saccharomyces_cerevisiae,/home/ubuntu/environment/github/phylorthology/assets/nf-test/test-data/Saccharomyces_cerevisiae-test-proteome.fasta,Opisthokonta,ascomycota_odb10,eukaryota_odb10,proteins,true,true
```

Comment:
Is it always best practice to have a trailing newline at the end of each file?

Reply:
yes

Reply:
1. If you use VSCode there is a plugin to add EOF new line automatically so you don't have to think about it - it should be in an AUG recording that Mert gave about developer productivity 
2. However in this file specifically will it mess things up if an empty newline is given for an input samplesheet?

Reply:
Okay - I think there was one case or two in things ive done in the past where I needed one to not be there and have in some cased started getting rid of them, but I can not do that any more! I don't know whether it would cause any problems here though!